aptos-move/move-examples/post_mint_reveal_nft/sources/minting.move

```diff
@@ -160,165 +177,147 @@ module post_mint_reveal_nft::minting {
         });
     }
 
-    /// Set admin of this module.
-    public entry fun set_admin(admin: &signer, new_admin_address: address) acquires NFTMintConfig {
-        let nft_mint_config = borrow_global_mut<NFTMintConfig>(@post_mint_reveal_nft);
-        assert!(signer::address_of(admin) == nft_mint_config.admin, error::permission_denied(ENOT_AUTHORIZED));
-        nft_mint_config.admin = new_admin_address;
-    }
-
     /// Set the treasury account (where the payment for NFT goes to) of this module.
     public entry fun set_treasury(admin: &signer, new_treasury_address: address) acquires NFTMintConfig {
+        assert!(signer::address_of(admin) == @post_mint_reveal_nft, error::permission_denied(ENOT_AUTHORIZED));
         let nft_mint_config = borrow_global_mut<NFTMintConfig>(@post_mint_reveal_nft);
-        assert!(signer::address_of(admin) == nft_mint_config.admin, error::permission_denied(ENOT_AUTHORIZED));
         nft_mint_config.treasury = new_treasury_address;
     }
 
-    /// Set up and create the destination collection.
+    /// Set up the source collection and destination collection.
+    /// If we have already called this function before, calling it again
+    /// will update the source and destination collection config.
     public entry fun set_collection_config_and_create_collection(
         admin: &signer,
-        collection_name: String,
-        collection_description: String,
-        collection_maximum: u64,
-        collection_uri: String,
-        collection_mutate_config: vector<bool>,
-        token_name_base: String,
+        source_collection_name: String,
+        source_token_name_base: String,
+        source_token_uri: String,
+        destination_collection_name: String,
+        destination_collection_uri: String,
+        destination_collection_maximum: u64,
+        destination_collection_description: String,
+        destination_token_name_base: String,
+        destination_collection_mutate_config: vector<bool>,
         royalty_payee_address: address,
         token_description: String,
         token_maximum: u64,
         token_mutate_config: vector<bool>,
         royalty_points_den: u64,
         royalty_points_num: u64,
         public_mint_limit: u64,
-    ) acquires NFTMintConfig {
-        let nft_mint_config = borrow_global_mut<NFTMintConfig>(@post_mint_reveal_nft);
-        assert!(signer::address_of(admin) == nft_mint_config.admin, error::permission_denied(ENOT_AUTHORIZED));
-
-        assert!(vector::length(&collection_mutate_config) == 3 && vector::length(&token_mutate_config) == 5, error::invalid_argument(EVECTOR_LENGTH_UNMATCHED));
+    ) acquires NFTMintConfig, CollectionConfig, SourceToken {
+        assert!(signer::address_of(admin) == @post_mint_reveal_nft, error::permission_denied(ENOT_AUTHORIZED));
+        assert!(vector::length(&destination_collection_mutate_config) == 3 && vector::length(&token_mutate_config) == 5, error::invalid_argument(EVECTOR_LENGTH_UNMATCHED));
```

Comment:
Is this not being verified by token.move itself?

Reply:
yes it is, but thought we should verify it and give it a more explicit error in minting.move

Reply:
IMO the error from token.move is good enough. But up to you - I don't feel strongly